tests/components/dlink/test_config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Test D-Link Smart Plug config flow."""
+from unittest.mock import MagicMock, patch
+
+from homeassistant import data_entry_flow
+from homeassistant.components.dlink.const import DEFAULT_NAME, DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.core import HomeAssistant
+
+from .conftest import CONF_DATA, CONF_IMPORT_DATA, patch_config_flow
+
+from tests.common import MockConfigEntry
+
+
+def _patch_setup_entry():
+    return patch("homeassistant.components.dlink.async_setup_entry")
+
+
+async def test_flow_user(hass: HomeAssistant, mocked_plug: MagicMock) -> None:
+    """Test user initialized flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+    with patch_config_flow(mocked_plug), _patch_setup_entry():
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input=CONF_DATA,
+        )
+    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+    assert result["title"] == DEFAULT_NAME
+    assert result["data"] == CONF_DATA
+
+
+async def test_flow_user_already_configured(
+    hass: HomeAssistant, config_entry: MockConfigEntry
+) -> None:
+    """Test user initialized flow with duplicate server."""
+    result = await hass.config_entries.flow.async_init(
```

Comment:
Add a MockConfigEntry matching the same user input instead.

Reply:
It looks like that is already being done. Am I not supposed to use a fixture here?

Reply:
Sorry, I think I read the same line twice.